File: business-backend/app/schemas/style.py
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime

from app.schemas.category import Category


# 基础风格模型
class StyleBase(BaseModel):
    name: str
    description: Optional[str] = None
    preview_url: Optional[str] = None
    reference_image_url: Optional[str] = None
    category_id: Optional[int] = None
    prompt: Optional[str] = None
    credits_cost: int = 10
    sort_order: int = 0


# 创建风格时的数据模型
class StyleCreate(StyleBase):
    category: Optional[str] = None


# 更新风格时的数据模型
class StyleUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    preview_url: Optional[str] = None
    reference_image_url: Optional[str] = None
    category_id: Optional[int] = None
    prompt: Optional[str] = None
    credits_cost: Optional[int] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


# 返回给API的风格模型
class Style(StyleBase):
    id: int
    is_active: bool
    created_at: datetime
    updated_at: datetime
    category_info: Optional[Category] = Field(None, validation_alias='category')

    class Config:
        orm_mode = True
        allow_population_by_field_name = True 

File: admin-frontend/src/api/style.ts
import request from '@/utils/request'

// 类型定义
export interface Style {
  id: number
  name: string
  description?: string
  preview_url?: string
  reference_image_url?: string
  category?: string
  category_id?: number
  prompt?: string
  credits_cost: number
  is_active: boolean
  sort_order?: number
  created_at: string
  updated_at: string
}

export interface StyleCreate {
  name: string
  description?: string
  preview_url?: string
  reference_image_url?: string
  category?: string
  category_id?: number
  prompt?: string
  credits_cost?: number
  sort_order?: number
}

export interface StyleUpdate {
  name?: string
  description?: string
  preview_url?: string
  reference_image_url?: string
  category?: string
  category_id?: number
  prompt?: string
  credits_cost?: number
  is_active?: boolean
  sort_order?: number
}

// API响应类型
export interface ApiResponse<T> {
  items: T[]
  total: number
}

// 获取风格列表
export function getStyles(params?: {
  page?: number
  limit?: number
  name?: string
  category?: string
  category_id?: number
  is_active?: boolean
}) {
  return request({
    url: '/styles',
    method: 'get',
    params
  })
}

// 获取单个风格详情
export function getStyle(id: number) {
  return request({
    url: `/styles/${id}`,
    method: 'get'
  })
}

// 创建风格
export function createStyle(data: StyleCreate) {
  return request({
    url: '/styles',
    method: 'post',
    data
  })
}

// 更新风格
export function updateStyle(id: number, data: StyleUpdate) {
  return request({
    url: `/styles/${id}`,
    method: 'put',
    data
  })
}

// 删除风格
export function deleteStyle(id: number) {
  return request({
    url: `/styles/${id}`,
    method: 'delete'
  })
}
